File: fe/src/api/system/frontend-route.ts
import request from '@/utils/request';

export interface IFrontendRoute {
    id?: number
    name: string;
    title: string;
    component_path: string;
    route_path: string;
}

// 创建前端路由
export const addApi = (data: IFrontendRoute): IRespData<null> => request.post("/FrontendRoute/Add", data)
// 编辑前端路由
export const editApi = (data: Required<IFrontendRoute>): IRespData<null> => request.post("/FrontendRoute/Edit", data)
// 删除前端路由
export const deleteApi = (id: number): IRespData<null> => request.post("/FrontendRoute/Delete", { id })
// 根据 ID 获取前端路由
export const getApi = (id: number): IRespData<Required<IFrontendRoute>> => request.get("/FrontendRoute/Get", { params: { id } })
// 分页查询前端路由
export const listPageApi = (data: IPager): IRespData<IPageData<Required<IFrontendRoute>>> => request.post("/FrontendRoute/ListPage", data)
// 查询所有前端路由
export const listAllApi = (): IRespData<Required<IFrontendRoute>[]> => request.get("/FrontendRoute/ListAll")

File: app/models/view/system/FrontendRoute.go
package v_system

type AddFrontendRouteReq struct {
	Name          string `json:"name" valid:"required~路由名称不能为空"`
	Title         string `json:"title" valid:"required~密码不能为空"`
	ComponentPath string `json:"component_path" valid:"required~组件路径不能为空"`
	RoutePath     string `json:"route_path" valid:"required~路由路径不能为空"`
}

type EditFrontendRouteReq struct {
	ID int `json:"id,optional" valid:"int~路由ID必须为整型数字,gt(0)~路由ID必须大于0"`
	AddFrontendRouteReq
}
